compiler/rustc_mir/src/interpret/eval_context.rs

```diff
@@ -852,11 +851,16 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         // Normal return, figure out where to jump.
         if unwinding {
             // Follow the unwind edge.
-            let unwind = next_block.expect("Encountered StackPopCleanup::None when unwinding!");
-            self.unwind_to_block(unwind);
+            let unwind = match return_to_block {
+                StackPopCleanup::Goto { unwind, .. } => unwind,
+                StackPopCleanup::None { .. } => {
+                    panic!("Encountered StackPopCleanup::None when unwinding!")
+                }
+            };
+            self.unwind_to_block(unwind)?;
         } else {
             // Follow the normal return edge.
-            if let Some(ret) = next_block {
+            if let StackPopCleanup::Goto { ret, .. } = return_to_block {
```

Comment:
Please write this as a `match`, so that if we ever add another variant to `StackPopCleanup` we don't forget to adjust this code.